src/lib/ui/SupportedDevices/SupportedDevicesTable.tsx

```diff
@@ -0,0 +1,149 @@
+import { useQuery } from '@tanstack/react-query'
+import axios from 'axios'
+import { useCallback, useEffect, useState } from 'react'
+
+import type { SupportedDevicesProps } from 'lib/index.js'
+
+import { Button } from 'lib/ui/Button.js'
+import { SupportedDevicesFilterArea } from 'lib/ui/SupportedDevices/SupportedDevicesFilterArea.js'
+import type { DeviceModel, Filters } from 'lib/ui/SupportedDevices/types.js'
+
+import { SupportedDeviceRow } from './SupportedDeviceRow.js'
+import { SupportedDevicesHeader } from './SupportedDevicesHeader.js'
+
+const BASE_URL = 'https://devicedb.seam.co/api/device_models/list'
+
+export interface SupportedDevicesTableProps {
+  cannotFilter?: boolean
+}
+
+export function SupportedDevicesTable({ cannotFilter }: SupportedDevicesTableProps) {
+  const [allDeviceModels, setAllDeviceModels] = useState<DeviceModel[]>([])
+  const [filterValue, setFilterValue] = useState('')
+  const [filters, setFilters] = useState<Filters>({
+    supportedOnly: false,
+    category: null,
+    brand: null,
+  })
+
+  const fetchDeviceModels = useCallback(async () => {
+    const queries = []
+
+    if (filterValue.trim() !== '') {
+      queries.push(`text_search=${encodeURIComponent(filterValue.trim())}`)
+    }
+
+    if (filters.supportedOnly) {
+      queries.push('support_level=live')
+    }
+
+    if (filters.category !== null) {
+      queries.push(`main_category=${encodeURIComponent(filters.category)}`)
+    }
+
+    if (filters.brand !== null) {
+      queries.push(`brand=${encodeURIComponent(filters.brand)}`)
+    }
+
+    const url = `${BASE_URL}?${queries.join('&')}`
+    return await axios.get(url)
+  }, [filterValue, filters])
+
+  const { data, isLoading, isError, refetch } = useQuery<{
+    data: {
+      device_models?: DeviceModel[]
+    }
+  }>({
+    queryKey: ['supported_devices', filterValue, filters],
+    queryFn: fetchDeviceModels,
+  })
+
+  useEffect(() => {
+    if (
+      data?.data?.device_models !== undefined &&
+      allDeviceModels.length === 0
+    ) {
+      setAllDeviceModels(data.data.device_models)
+    }
+  }, [data, allDeviceModels.length])
+
+  const deviceModels = data?.data?.device_models ?? []
+
+  const showFilterArea = cannotFilter == null ? true : !cannotFilter
```

Comment:
Personally, I find this much more confusing than simply `!Boolean(cannotFilter)` or just `!cannotFilter`. We might want to consider disabling the `@typescript-eslint/strict-boolean-expressions` linting rule because of how often it causes hiccups

Reply:
Okay. Updated the props to make it default to `false`